### ISSUE TITLE
CertifyVuln should be scanner/db-agnostic

### ISSUE THREAD

**Issue by bobmcwhirter** [guacsec/guac#1097]
Currently CertifyVuln and IsVulnerability are somewhat intertwingly and presume a closed-world existing of OSV, CVE and GHSA.

While OSV ultimately is backed by many other datasources, GUAC should either explicitly support zero ecosystems, or absolutely every ecosystem. 

Instead of maintaining an every-increasing set of unioned types for every new scanner/db/ecosystem that provides vulnerability information, move towards using `Vulnerability URLs` or `vURLs` (invented here, not an industry term).

Collectors could exist and connect via collectsub-like facilities, handling any recognizable vURL that choose to.

General format could be `vuln://<general-type>/<specific-details>`. I suspect we could forbid qualifiers/parameters, and require exactly a path.

Examples:

CVE, using path separator: `vuln://cve/2023/8675`
OSV, representing its knowledge of a GHSA: `vuln://osv/GHSA-205hk`
Random vendor: `vuln://snyk/SN-whatever`

**Comment by pxp928** [comment]
Thanks @bobmcwhirter for opening the issue. This is a good start for the `vURLs` format. For the `<specific-details>`, I think we should make it a standard format that all can follow. Just having it be the vulnerability ID (or vendor vulnerability ID) would make sense. Leaving it ambiguous might bite us later when we want to parse.

This could follow the similar "trie" representation that we use for packages and sources. Allowing for various types to be created in the future without having to create new node types.

**Comment by bobmcwhirter** [comment]
Yah, I only mentioned year for CVE due to the fact GUAC is already treating it special, somewhat.

**Comment by pxp928** [comment]
Yeah, I don't recall the use case we had for adding `year` as a separate field to the CVE node. Can be removed as there is nothing it is currently used for.

**Comment by mihaimaruseac** [comment]
+1 to having a purl-like schema for vulnerabilities.

I think we wanted to be able to retrieve all CVEs within a single year at some point (for statistics maybe?) but I think this should be pushed out to the user code

**Comment by bobmcwhirter** [comment]
It should be a vURI not vURL. 

https://hachyderm.io/@bobmcwhirter/110772282108669965

**Comment by bobmcwhirter** [comment]
You rock, thanks!

**Comment by pxp928** [comment]
work has started for this and will go into https://github.com/guacsec/guac/tree/vuln-gql-change branch for the time being (as it is a breaking change)
